src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -93,6 +96,18 @@ protected virtual void InitializeAssembly (AssemblyDefinition assembly)
 			}
 		}
 
+		void Complete ()
+		{
+			bool anyConverted = false;
+			foreach (var body in _pendingBodies) {
+				anyConverted = true;
+				Annotations.SetAction (body.Method, MethodAction.ConvertToThrow);
+			}
+
+			if (anyConverted)
+				MarkAndCacheNotSupportedCtorString ();
```

Comment:
I think this is in theory too late and should be called during Process. Call it in inside MarkMethodBody

Reply:
Good catch.